From eebf7ae0bc8d3dcfbdf0d281872ae268d0316bc4 Mon Sep 17 00:00:00 2001
From: Carlos Alberto Lopez Perez <clopez@igalia.com>
Date: Tue, 19 Dec 2017 13:45:31 +0100
Subject: [PATCH 6/6] Return wrong allocator instead of asserting

---
 gst-libs/gst/gl/gstglbufferpool.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglbufferpool.c b/gst-libs/gst/gl/gstglbufferpool.c
index 53d3b66d9..c60f62c74 100644
--- a/gst-libs/gst/gl/gstglbufferpool.c
+++ b/gst-libs/gst/gl/gstglbufferpool.c
@@ -126,7 +126,8 @@ gst_gl_buffer_pool_set_config (GstBufferPool * pool, GstStructure * config)
   } else {
     priv->allocator =
         GST_ALLOCATOR (gst_gl_memory_allocator_get_default (glpool->context));
-    g_assert (priv->allocator);
+    if (!priv->allocator)
+      goto wrong_allocator;
   }
 
   priv->add_videometa = gst_buffer_pool_config_has_option (config,
-- 
2.11.0

